File: src/main/java/pl/mjedynak/idea/plugins/builder/factory/GenerateBuilderPopupListFactory.java
package pl.mjedynak.idea.plugins.builder.factory;

import static java.util.Arrays.asList;

import com.intellij.ui.components.JBList;
import org.jetbrains.annotations.NotNull;
import pl.mjedynak.idea.plugins.builder.action.AbstractBuilderAdditionalAction;
import pl.mjedynak.idea.plugins.builder.action.GoToBuilderAdditionalAction;
import pl.mjedynak.idea.plugins.builder.action.RegenerateBuilderAdditionalAction;

public class GenerateBuilderPopupListFactory extends AbstractPopupListFactory {
    @Override
    protected @NotNull JBList<AbstractBuilderAdditionalAction> createList() {
        return new JBList<>(asList(new GoToBuilderAdditionalAction(), new RegenerateBuilderAdditionalAction()));
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/factory/GenerateBuilderPopupListFactory.kt
package io.github.kingg22.kobra.builder.factory

import com.intellij.ui.components.JBList
import io.github.kingg22.kobra.builder.action.AbstractBuilderAdditionalAction
import io.github.kingg22.kobra.builder.action.GoToBuilderAdditionalAction
import io.github.kingg22.kobra.builder.action.RegenerateBuilderAdditionalAction

public class GenerateBuilderPopupListFactory : AbstractPopupListFactory() {
    override fun createList(): JBList<AbstractBuilderAdditionalAction> =
        JBList(GoToBuilderAdditionalAction(), RegenerateBuilderAdditionalAction())
}
